File: Algorithms/17-merge-sorted-array/Another.c

void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n){
    for(int i=m+n-1,j=m-1,k=n-1; i>=0; i--)
                if(k<0) 
                    nums1[i]=nums1[j--];
                else if(j<0) 
                    nums1[i]=nums2[k--];
                else if(nums1[j]>nums2[k])
                    nums1[i]=nums1[j--];
                else
                    nums1[i]=nums2[k--];
}

File: Algorithms/17-merge-sorted-array/Another.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    var j int = m-1
    var k int = n-1
    for i := m+n-1; i>=0; i-- {
        if k<0 {
            nums1[i]=nums1[j]
            j=j-1
        } else if j<0 {
            nums1[i]=nums2[k]
            k=k-1
        } else if nums1[j]>nums2[k] {
            nums1[i]=nums1[j]
            j=j-1
        } else{
            nums1[i]=nums2[k]
            k=k-1
        }
    }
            
}